### PR TITLE
bpo-37073: clarify docs about PyBytes_FromStringAndSize and io.readinto

### PR DESCRIPTION
**Pull request by Windsooon**
https://bugs.python.org/issue37073

### CODE REVIEW

File: Doc/library/io.rst [hunk 1]
```diff
@@ -403,9 +403,10 @@ I/O Base Classes
 
       Read bytes into a pre-allocated, writable
       :term:`bytes-like object` *b*, and return the
-      number of bytes read.  For example, *b* might be a :class:`bytearray`.
-      If the object is in non-blocking mode and no bytes
-      are available, ``None`` is returned.
+      number of bytes that read succeed, it may be smaller
```
Comment (auvipy) [inline Doc/library/io.rst:406]: no need of it before may

**Review by auvipy — APPROVED** [review]

File: Doc/library/io.rst (hunk 1 above)
Comment (Windsooon) [reply Doc/library/io.rst:406]: Are you suggest we should delete 'it' before `may be smaller...`?

File: Doc/c-api/bytes.rst [hunk 2]
```diff
@@ -43,9 +43,9 @@ called with a non-bytes parameter.
 
 .. c:function:: PyObject* PyBytes_FromStringAndSize(const char *v, Py_ssize_t len)
 
-   Return a new bytes object with a copy of the string *v* as value and length
-   *len* on success, and *NULL* on failure.  If *v* is *NULL*, the contents of
-   the bytes object are uninitialized.
+   Return a new bytes object with a copy of the first *len* bytes of string *v* as
+   value and length *len* on success, and *NULL* on failure.  If *v* is *NULL*, the
```
Comment (methane) [inline Doc/c-api/bytes.rst:47]: It seems redundant to me.
Current "copy of the string *v* as value and length *len*" is understandable.
And "copy of first *len* bytes of the string *v*" also make sense.

File: Doc/library/io.rst [hunk 3]
```diff
@@ -403,9 +403,10 @@ I/O Base Classes
...
+      than *b* or NULL if failed.  For example, *b* might
+      be a :class:`bytearray`.  If the object is in non-blocking
+      mode and no bytes are available, ``None`` is returned.
```
Comment (methane) [inline Doc/library/io.rst:409]: I prefer current version to your requested version.
But since I'm not good English writer, please wait another review.

**Comment by csabella** [comment]
@Windsooon , please resolve the merge conflict and also take a look at the code review from @methane .  Thank you!

**Comment by Windsooon** [comment]
It looks @methane prefers the current version instead of my own version. Maybe I should close the PR?

**Comment by methane** [comment]
I replied in bpo.

**Pull request closed without merging**